x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/RollupJobConfig.java

```diff
@@ -128,7 +137,7 @@ public RollupJobConfig(final String id,
         this.indexPattern = indexPattern;
         this.rollupIndex = rollupIndex;
         this.groupConfig = groupConfig;
-        this.metricsConfig = metricsConfig != null ? metricsConfig : Collections.emptyList();
+        this.metricsConfig = addDefaultMetricsIfNeeded(metricsConfig, groupConfig);
```

Comment:
Since we always guarantee at least one metric is included (either explicitly, or through default), I think we can get rid of the `"At least one grouping or metric must be configured"` check a little bit above this line.  That was just to make sure users didn't configure a completely empty Rollup.  

I think that's moot now that we include some defaults, and an empty rollup with those defaults might still be useful (min/max of histos per bucket, etc).

We can also remove the null checks on `metricsConfig` in the `getAllFields()` and `toXContent()` methods I think?

Reply:
@polyfractal if both are null, then it will still be an empty rollup as there would be no groupings from which to derive default metrics.

Reply:
We enforce that a date_histo is always configured, so at a minimum there will be buckets of time.

We can leave the check for now, it might not be useful to just have time buckets + min/max time.  Probably easier to remove later if a user needs it, rather than allow it and regret it later :)

Reply:
Yes, we require a `date_histo` in the `GroupConfig` object, but if that config is null, then we have nothing from which to gather default values. 

I think if the desire is for a date_histo to ALWAYS be provided, there may be a bug. If `GroupConfig` is null, but a `MetricConfig` objects are provided, then that null check passes and no date_histogram configuration is provided. 

Additionally `validateMappings` would throw a NPE if the groupConfig was null.

Reply:
Yeah I think you're right, that's an NPE waiting to happen.  Let's ignore that for now in this PR to keep the changes minimal... I (or you if you want! ;) ) can followup with a bugfix for that null groupconfig issue.